### ISSUE TITLE
make check failed if I rewrite the unk value

### ISSUE THREAD

**Issue by deasuke** [google/sentencepiece#31]
This is rather a feature request than an issue.

I want to reserve small word IDs for special purpose and so I set the unk ID to 3.
const uint32 ModelInterface::kUnkID = 3;

The code was built, however,  make check failed.

**Comment by deasuke** [comment]
Is that an only test-case problem or it needs more effort?

**Comment by taku910** [comment]
BOS/EOS/UNK ids are hard coded. Not sure what happens when rewriting these variables.

By the way, If you want to reserve some ids, you can train sentencepiece with --contol_symbols flag
https://github.com/google/sentencepiece/blob/master/doc/special_symbols.md
--control_symbols="<my_tag1>,<my_tag2>"

then, "<my_tag1>" and "<my_tag2>" are reserved. These vocab must be inserted outside of sentencepiece. The id of these symbols can be found in vocab file.

**Comment by deasuke** [comment]
Thanks for the advice. However, I'd like to reserve symbol IDs, but not names....

**Comment by deasuke** [comment]
Here's my fork with (bos=0, eos=1, unk=2) for who is interested in this; https://github.com/deasuke/sentencepiece

A lot of sequence processing system use pad=0 and eos=1 and has no bos; e.g. tf.reduce_sum(some_position_slice)==0 means every word in the position is PAD and tf.less(some_position_slice, 2) means "choose finished sequences"

BTW, I have a question: I have to rewrite the test case of "我輩は猫である" (see unigram_trainer_test.cc of the change https://github.com/deasuke/sentencepiece/commit/4a9f0414ee09e95870335314599ec482ed9ccdb3  )
Is the algorithm dependent on (the order of) word ID ?

**Comment by hidetokazawa** [comment]
This feature may be hard to implement because some IDs are hard-coded but it will be helpful when using SentencPiece with [Tensor2Tensor](https://github.com/tensorflow/tensor2tensor) because it uses some ids (0 and 1) for special purposes. I imagine there are other cases when downstream programs reserves some ids whose meanings are different from SentencePiece.

**Comment by taku910** [comment]
Updated the code such that ids for <unk> <s> </s> are redefined.

% spm_train --bos_id=0 --eos_id=1 --unk_id=2 ....

Also id for padding <pad> is added, though it is disabled by default. Will update the document soon.

https://github.com/google/sentencepiece/commit/d1028974960d9e7ac9b408f6c212aa90d7c958c
